### DISCUSSION TITLE
Use non priv ports for dnsmasq in container version of pi-hole?

### DISCUSSION THREAD

**Discussion by zestysoft** [pi-hole/pi-hole#4678] [Bugs]
Is it possible to set the pi-hole container to listen on non-privileged ports for dns tcp/udp traffic?  I've been running pi-hole in a Kubernetes cluster using the CRIO container engine with the SecComp feature gate enabled, and found that the latest docker version 2022.04 was flapping with this in the logs and never fully starting:
```
Starting pihole-FTL (no-daemon) as pihole
Stopping pihole-FTL
pihole-FTL: no process found
Starting pihole-FTL (no-daemon) as pihole
Stopping pihole-FTL
pihole-FTL: no process found
```
After getting a shell into the container and running the diagnostics, I saw this in the logs:
```
[2022-04-02 19:27:48.675 3294M] FATAL ERROR in dnsmasq core: failed to create listening socket for port 53: Permission denied
   [2022-04-02 19:27:48.785 3294M] ########## FTL terminated after 446ms  (code 1)! ##########
```

If I roll back to 2022.01.01 it fires up fine, but that got me thinking that since Kubernetes uses services and ingresses to communicate with processes in containers anyway, there really isn't a need to run anything in a container on a privileged port for my use case (I don't use pi-hole for the dhcp stuff).

**Comment by bkeitht** [comment]
I was able to add the following and can now use the latest version of pihole.

        image: pihole/pihole:latest
        imagePullPolicy: Always
        **env:
        - name: DNSMASQ_USER
          value: "root"**
